pkg/model/components/containerd.go

```diff
@@ -61,26 +61,58 @@ func (b *ContainerdOptionsBuilder) BuildOptions(o interface{}) error {
 			}
 		}
 
+		// Throw a warning if containerd configoverride is populated in the cluster spec
+		klog.Warning("Overriding the configuration of containerd has to be done with care as the default config may" +
+			" change with releases which can lead to incompatibilities.")
+
 		// Apply defaults for containerd running in container runtime mode
 		containerd.LogLevel = fi.String("info")
 		usesKubenet := UsesKubenet(clusterSpec.Networking)
+		lines := []string{}
 		if clusterSpec.Networking != nil && usesKubenet {
 			// Using containerd with Kubenet requires special configuration. This is a temporary backwards-compatible solution
 			// and will be deprecated when Kubenet is deprecated:
 			// https://github.com/containerd/cri/blob/master/docs/config.md#cni-config-template
-			lines := []string{
+			lines = []string{
 				"version = 2",
 				"[plugins]",
 				"  [plugins.\"io.containerd.grpc.v1.cri\"]",
 				"    [plugins.\"io.containerd.grpc.v1.cri\".cni]",
 				"      conf_template = \"/etc/containerd/cni-config.template\"",
 			}
-			contents := strings.Join(lines, "\n")
-			containerd.ConfigOverride = fi.String(contents)
+
+			// Use systemd as the default cgroup-driver for containerd for clusters running k8s 1.20
+			if b.IsKubernetesGTE("1.20") {
+				systemdConfig := []string{
+					"  [plugins.\"io.containerd.grpc.v1.cri\".containerd]",
+					"    [plugins.\"io.containerd.grpc.v1.cri\".containerd.runtimes]",
+					"      [plugins.\"io.containerd.grpc.v1.cri\".containerd.runtimes.runc]",
+					"        runtime_type = \"io.containerd.runc.v2\"",
+					"          [plugins.\"io.containerd.grpc.v1.cri\".containerd.runtimes.runc.options]",
+					"            SystemdCgroup = true\n",
+				}
+				lines = append(lines, systemdConfig...)
+			}
 		} else {
-			containerd.ConfigOverride = fi.String("")
+			lines = []string{
+				"version=2",
+				"[plugins]",
+			}
+			// Use systemd as the default cgroup-driver for containerd for clusters running k8s 1.20
+			if b.IsKubernetesGTE("1.20") {
+				systemdConfig := []string{
+					"  [plugins.\"io.containerd.grpc.v1.cri\".containerd]",
+					"    [plugins.\"io.containerd.grpc.v1.cri\".containerd.runtimes]",
+					"      [plugins.\"io.containerd.grpc.v1.cri\".containerd.runtimes.runc]",
+					"        runtime_type = \"io.containerd.runc.v2\"",
+					"          [plugins.\"io.containerd.grpc.v1.cri\".containerd.runtimes.runc.options]",
```

Comment:
As you're switching to the default CRI plugin runtime `io.containerd.runc.v2` for Containerd 1.4, the `SystemdCgroup` option can be removed as it's only relevant for `io.containerd.runtime.v1.linux`.

https://github.com/containerd/containerd/blob/master/pkg/cri/config/config.go#L213-L216